docs/user-guide/rule-deprecation.md

```diff
@@ -0,0 +1,14 @@
+# Rule Deprecation
+
+Balancing the trade-offs of improving a tool and the frustration these changes can cause is a difficult task. One key area in which this affects our users is in the removal of rules.
+
+The ESLint team is committed to making upgrading as easy and painless as possible. To that end, the team has agreed upon some guidelines for deprecating rules in the future. The goal of these guidelines is to allow for improvements and changes to be made without breaking existing configurations.
+
+Until May 1, 2017, the team has committed to not remove any rules in any releases of ESLint; however, the team will deprecate rules as needed. When a rule is deprecated, it means that:
+
+* The team will no longer do any work on the rule. This includes bug fixes, enhancements, and updates to the rule's documentation. Issues and pull requests related to the deprecated rule will not be accepted and will be closed automatically.
+* The rule is moved to a list of deprecated rules rather than being on the main rule page. Additionally, the rule will be labeled as deprecated on the rule's documentation page.
```

Comment:
That is flexible without being vague. Sorry that I have not yet thought enough about information people will need in various situations: how it will look from the outside-in instead of inside-out.

From a quick look at history of the rules index page, we have done the following at times:

1. kept rule in the main list for a while with a **(deprecated)** label
2. moved rules to a **Deprecated** section
3. moved rules to a **Removed** section

The alternative wording still commits **never again** to do 1 and defers decision whether ever to do 2 (in analogy with 3) or move the rules elsewhere immediately.